internal/mode/static/nginx/agent/agent.go

```diff
@@ -0,0 +1,33 @@
+package agent
+
+import "fmt"
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+//counterfeiter:generate . NginxUpdater
+
+// NginxUpdater is an interface for updating NGINX using the NGINX agent.
+type NginxUpdater interface {
+	UpdateNginxConfig(int)
```

Comment:
nit: stutters a little
```suggestion
	UpdateConfig(int)
```